app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -452,6 +470,9 @@ class FormController(
       def processSaveAndExit(data: FormDataRecalculated, sections: List[Section]): Future[Result] =
         validateAndUpdateData(data, sections) { maybeSn =>
           val formTemplate = cache.formTemplate
+          val formatter = DateTimeFormatter.ofPattern("dd MMMM yyyy")
+          val envelopeExpiryDate =
+            cache.form.envelopeExpiryDate.fold("for 30 days")("until " + _.ldt.format(formatter))
```

Comment:
This needs to be done in the view, where it will be amenable to welsh translation, etc..